src/ecdsa/rfc6979.py

```diff
@@ -56,12 +56,13 @@ def bits2octets(data, order):
 
 
 # https://tools.ietf.org/html/rfc6979#section-3.2
-def generate_k(order, secexp, hash_func, data):
+def generate_k(order, secexp, hash_func, data, fail_loops=0):
     '''
         order - order of the DSA generator used in the signature
         secexp - secure exponent (private key) in numeric form
         hash_func - reference to the same hash function used for generating hash
         data - hash in binary form of the signing data
+        fail_loops - integer of the number of times to loop due to failure
```

Comment:
it's not really a failure on `generate_k`'s part, and I think this makes is rather confusing

it's more like "retry_gen - int - how many good 'k' values to skip before returning"